### PR TITLE
Feature/limited rights

### PR DESCRIPTION
**Pull request by J4bbi**
* Issue: #75

---

This PR limits the availability of licenses for the user to choose from in the submission form. The full list of 419 licenses is taken from the [SPDX list of licenses](https://spdx.org/licenses/) is to be found in [invenio-rdm-records](https://github.com/inveniosoftware/invenio-rdm-records/blob/8d2823f3617873511b67e4c3739ebb381e474a97/invenio_rdm_records/fixtures/data/vocabularies/licenses.csv).

The standard functionality is to allow multiple licenses and custom licenses. This is done by providing an "Add custom" button, as well as an "Add standard" button. Clicking the "Add standard" button open ups a search modal that searches the `api/vocabularies/licenses` endpoint using the [react-searchkit](https://github.com/inveniosoftware/react-searchkit) library.

This PR creates three new/modifies components: `LimitedLicenseFieldItem`, `LimitedLicenseModal` and `LimitedLicenseField`.

- The Add custom button is omitted
- The Add standard button is renamed to Add license
- The Add license button is only displayed if the number of licenses is falsy
- The name of the Edit button of an already selected license is renamed to Change
- The search bar and the filtering by tags (Recommended, All, Data and Software) are omitted
- The initialQueryState of the react-searchkit searchConfig is filter by the data tag, given a query to search by id and limited to two results

The backend endpoint is of course untouched.

## Developer Checklist

*Developers should review and confirm each of these items before requesting review*

* [ ] Code meets acceptance criteria from issue
* [ ] Unit tests are written and all pass
* [ ] User Test Scripts (if required) are written and have been run through
* [ ] Code documentation and related non-code documentation has all been updated

## Reviewer Checklist

*Reviewers should review and confirm each of these items before approval*
*If there are multiple reviewers, this section can be duplicated for each reviewer*

* [ ] Code meets acceptance criteria from issue
* [ ] Unit tests are written and all pass
* [ ] User Test Scripts (if required) are written and have been run through
* [ ] Code documentation and related non-code documentation has all been updated
* [ ] Migation has been created and tested

## Testing

*List user test scripts that need to be run*

*List any non-unit test scripts that need to be run*

### CODE REVIEW

**Review by cc-a — CHANGES_REQUESTED** [review]
Thanks @J4bbi. In terms of the resulting UI I think this is good. I'm slightly conflicted on the implementation tbh.

I can see that this was the shortest possible route to get from the existing license field to the behaviour we want but the end result is that for what we get (a modal with two buttons give or take) the code seems overly complex.

I'm aware that your guys time is finite so lets go ahead with this implementation for now (after addressing the comments I've added). I'm going to add an item to the backlog to explore simplifying the implementation down the line.

This is a potentially naive question but could a select input do the job?

File: site/ic_data_repo/assets/semantic-ui/js/ic_data_repo/LimitedLicenseModal.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    // we get a memory leak warning.
+    const { onLicenseChange } = this.props;
+    this.closeModal();
+    onLicenseChange(values.selectedLicense);
+    formikBag.resetForm();
+  };
+
+  render() {
+    const {
+      licenses,
```
[the remark this answers is not in the document]
Comment (J4bbi) [reply site/ic_data_repo/assets/semantic-ui/js/ic_data_repo/LimitedLicenseModal.js]: Very good point. I wasn't thinking clearly.

File: site/ic_data_repo/assets/semantic-ui/js/ic_data_repo/LimitedLicenseModal.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+  onSubmit = (values, formikBag) => {
+    // We have to close the modal first because onLicenseChange and passing
+    // license as an object makes React get rid of this component. Otherwise
+    // we get a memory leak warning.
+    const { onLicenseChange } = this.props;
+    this.closeModal();
+    onLicenseChange(values.selectedLicense);
+    formikBag.resetForm();
+  };
```
[the remark this answers is not in the document]
Comment (J4bbi) [reply site/ic_data_repo/assets/semantic-ui/js/ic_data_repo/LimitedLicenseModal.js]: Yes, certainly.

**Comment by J4bbi** [comment]
> Thanks @J4bbi. In terms of the resulting UI I think this is good. I'm slightly conflicted on the implementation tbh.
> 
> I can see that this was the shortest possible route to get from the existing license field to the behaviour we want but the end result is that for what we get (a modal with two buttons give or take) the code seems overly complex.
> 
> I'm aware that your guys time is finite so lets go ahead with this implementation for now (after addressing the comments I've added). I'm going to add an item to the backlog to explore simplifying the implementation down the line.
> 
> This is a potentially naive question but could a select input do the job?

Thanks @cc-a 

I found it tricky to replicate the functionality of the current component, that would also have required providing the hard-coded licenses and not use the search index.

**Review by cc-a — APPROVED** [review]
Thanks @J4bbi. Looks good.

**Comment by J4bbi** [comment]
The search criteria is defined in `LimitedLicenseField`:

https://github.com/ImperialCollegeLondon/fair-data-repository/blob/53249154781ee26b86d6734fb123f6e04039c3eb/site/ic_data_repo/assets/semantic-ui/js/ic_data_repo/LimitedLicenseField.js#L74-L78

**Pull request merged**